tests/callbacks/test_RecorderListener.cpp

```diff
@@ -0,0 +1,169 @@
+    );
+
+    ASSERT_THAT(
+        recorder->recorded_events,
+        ::testing::Contains(ABeginFenceEventWithName(::testing::StrEq("computation - level 0 - fence after pfor")))
+    );
+
+    ASSERT_THAT(
+        recorder->recorded_events,
+        ::testing::Contains(AProfileEventWithName(::testing::StrEq("buried marker")))
+    );
+
+    ASSERT_THAT(
+        recorder->recorded_events,
+        ContainsInOrder<typename decltype(recorder->recorded_events)::value_type>(
+            APushRegionEventWithName(::testing::StrEq("computation - level 0")),
+            ABeginParallelForEventWithName(::testing::StrEq("computation - level 0 - pfor")),
+            AEndParallelForEvent(),
+            ABeginFenceEventWithName(::testing::StrEq("computation - level 0 - fence after pfor")),
+            AEndFenceEvent(),
+            AProfileEventWithName(::testing::StrEq("buried marker")),
+            APushRegionEventWithName(::testing::StrEq("computation - level 1")),
+            ABeginParallelReduceEventWithName(::testing::StrEq("computation - level 1 - preduce on default exec")),
+            AEndParallelReduceEvent(),
+            ABeginFenceEventWithName(::testing::StrEq("Kokkos::parallel_reduce: fence due to result being value, not view")),
+            AEndFenceEvent(),
+            APopRegionEvent(),
+            APopRegionEvent()
+        )
+    );
+
+    ASSERT_GT(any_event_recorder->recorded_events.size(), recorder->recorded_events.size());
+
+    const auto fence_event_outside_profile_section = ABeginFenceEventWithName(::testing::StrEq("other fence after stopping the profile section"));
+
+    ASSERT_THAT(
+        recorder->recorded_events,
+        ::testing::Not(::testing::Contains(fence_event_outside_profile_section))
+    );
+
+    ASSERT_THAT(
+        any_event_recorder->recorded_events,
+        ::testing::Contains(fence_event_outside_profile_section)
+    );
+
+    const auto output = [&recorder] () -> std::string {
+        std::ostringstream oss;
+        recorder->report(oss);
+        return oss.str();
+    }();
+
+    ASSERT_THAT(output, ::testing::HasSubstr("computation - level 0 - pfor"));
+    ASSERT_THAT(output, ::testing::HasSubstr("computation - level 0 - fence after pfor"));
+
+    Manager::unregister_listener(recorder.get());
+    Manager::unregister_listener(any_event_recorder.get());
+}
+
+//! Listener to record fence events.
+using fence_finder_t = RecorderListener<EventRegexMatcher, BeginFenceEvent>;
+
+//! @test Check traits of @ref fence_finder_t.
+TEST(FenceFinder, traits)
+{
+    //! Check that @ref fence_finder_t is a listener for @ref Kokkos::utils::callbacks::BeginFenceEvent.
+    static_assert(std::same_as<
+        listener_event_type_list_t<fence_finder_t>,
+        Kokkos::Impl::type_list<BeginFenceEvent>
+    >);
+
+    static_assert(Listener<fence_finder_t>);
+}
+
+//! @test Check the behavior of @ref fence_finder_t.
+TEST(FenceFinder, recorded_events)
+{
+    Manager::initialize();
```

Comment:
Maybe worth introducing a macro for declaring a test that uses the ManagerTestFixture ?

Reply:
OK. Done.